packages/amplication-server/src/core/queue/queue.service.ts

```diff
@@ -5,16 +5,21 @@ export const QUEUE_SERVICE_NAME = "QUEUE_SERVICE";
 
 @Injectable()
 export class QueueService {
-  generatePullRequestTopic: string;
   constructor(
     @Inject(QUEUE_SERVICE_NAME)
     private readonly kafkaClient: ClientKafka
   ) {}
 
   async emitMessage(topic: string, message: string): Promise<void> {
-    // Explicitly wait for kafka client to connect. https://github.com/nestjs/nest/issues/10449
-    await this.kafkaClient.connect();
-
-    this.kafkaClient.emit(topic, message);
+    return await new Promise((resolve, reject) => {
+      this.kafkaClient.emit(topic, message).subscribe({
+        error: (err: any) => {
+          reject(err);
+        },
+        next: () => {
+          resolve();
+        },
+      });
```

Comment:
@EugeneTseitlin I saw your comment on the issue (https://github.com/amplication/amplication/issues/4947#issuecomment-1403765358) and I just want to make sure if this PR really fixes the issue or just catch the error and show it on screen

Reply:
@EugeneTseitlin can you get more details from the error in order to try  maybe re-emitting the event or checking the readiness of the kafkaClient before the emission.

Reply:
@yuval-hazaz the change solves issue as it was declared "Code generation never ends". So, code generation won't stuck forever in case that a message can't be sent to kafka, it will be marked as failed then.

Reply:
@overbit Promise already catches the whole error object, there are no other details. Retry is an option, but for now I've decided not to implement it in order to see how often this issue occurs. Now having logs it can be tracked.

Reply:
@EugeneTseitlin the issue describes this "Looking on the logs, we can see that the build message is sent to the Kafka queue, but by looking on kafka UI we never see the message" 

just to make sure - your fix does not resolve the root issue, it only resolves the issue that the UI is not updated.

What about the actual root cause? are we still looking at it?